File: src/main/java/com/key/oa/util/RedisUtil.java
package com.key.oa.util;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.redis.core.RedisTemplate;
import org.springframework.stereotype.Component;

import java.util.concurrent.TimeUnit;

/**
 * @author 孙强
 * 用于快捷地操作redis数据库
 */
@Component
public class RedisUtil {
    private final RedisTemplate<String, Object> redisTemplate;

    @Autowired
    public RedisUtil(RedisTemplate<String, Object> redisTemplate) {
        this.redisTemplate = redisTemplate;
    }

    public Boolean setExpire(String key, long seconds) {
        return redisTemplate.expire(key, seconds, TimeUnit.SECONDS);
    }

    public Long getExpire(String key) {
        return redisTemplate.getExpire(key, TimeUnit.SECONDS);
    }

    public void setKeyValue(String key, Object value) {
        redisTemplate.opsForValue().set(key, value);
    }

    public Object getValue(String key) {
        return redisTemplate.opsForValue().get(key);
    }

    public void deleteValue(String key) {
        redisTemplate.delete(key);
    }
}

File: src/main/kotlin/oa/util/RedisUtil.kt
package oa.util

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.data.redis.core.RedisTemplate
import org.springframework.stereotype.Component
import java.util.concurrent.TimeUnit

@Component
class RedisUtil @Autowired constructor(val redisTemplate: RedisTemplate<String, Any>) {
    fun setExpire(key: String, timeoutSeconds: Long): Boolean =
        redisTemplate.expire(key, timeoutSeconds, TimeUnit.SECONDS)

    fun getExpire(key: String): Long = redisTemplate.getExpire(key)

    fun setValue(key: String, value: Any): Unit = redisTemplate.opsForValue().set(key, value)

    fun setValue(key: String, value: Any, timeoutSeconds: Long): Unit =
        redisTemplate.opsForValue().set(key, value, timeoutSeconds, TimeUnit.SECONDS)

    fun getValue(key: String): Any? = redisTemplate.opsForValue().get(key)

    fun deleteValue(key: String): Boolean = redisTemplate.delete(key)
}